### DISCUSSION TITLE
Docker Dev env + git + pre-commit

### DISCUSSION THREAD

**Discussion by Bbillyben** [inventree/InvenTree#3749] [Q&A] [answered]
Hi there, 

I'm trying to commit to a fork of Inventree, from a docker dev setup on a debian VM.

Installation ok, and dev server running.

Git works fine, authentication ok, commit ok, etc.

Then I've ran : 
`docker compose run  inventree-dev-server setup-dev`

it seems working fine

then i made some modification to a file and run
```
git add path/to/file.py
docker compose run  inventree-dev-server pre-commit
```
and I can see all tests and their result (all ok for the next steps)

but then when i 
`git commit`
I've got the error : 
`pre-commit not found.  Did you forget to activate your virtualenv?`

of course I haven't any python package installed, every thing is packed in containers, so no venv.
I've try to only install pip3 && pre-commit, but same result.

My googling do not lead me to any sort of clues, and as you can imagine i'm a deep newbie on that process.

What Have I missed? Should I consider going for a baremetal install? or just install pyhton package and commit on a specific way ?

**Comment by wolflu05** [comment, marked as answer]
You get this error outside of the docker container where Python and everything is installed. While you did the setup dev thing, you installed the git hook pre commit, which expects to be run in an virtual env which lives in the container. But as you are outside of the container, you have not activated this venv (you even don't have python there, which is totally fine). For fixing this, you can commit with the `--no-verify` flag outside of the container, or exec into it, and run the commit there. As you already run the tests, you don't have to worry, that the precomit will find something.

**Reply by Bbillyben** [reply]
thanks for the tips @wolflu05 
I've tried toi commit from the container, but I should reconfigure all git credentials, implements new volume to store ssh keys, .... well the no-verify flag will do the tricks!

**Reply by wolflu05** [reply]
@Bbillyben Are you using vscode? Then I suggest you to use the devcontainers. You can find a WIP video here, which shows you how to setup. https://github.com/inventree/InvenTree/pull/3474#issuecomment-1207371811

**Reply by Bbillyben** [reply]
thks @wolflu05, I'm using visual studio && netbeans, but I will give a try on vs code, the video you quoted is really appealing!
